typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,592 @@
+                if (error === null) {
+                    resolve(okAsync(response));
+                } else {
+                    resolve(errAsync(error));
+                }
+            })
+        });
+        const resultRegisterService: Result<RegisterServiceResponse, Error> = await promiseRegisterService;
+        if (!resultRegisterService.isOk()) {
+            return err(resultRegisterService.error);
+        }
+        const registerServiceResp: RegisterServiceResponse = resultRegisterService.value;
+
+        const serviceIpAddr: string = registerServiceResp.getIpAddr();
+
+        const serviceContext: ServiceContext = new ServiceContext(
+            this.client,
+            serviceId,
+            serviceIpAddr,
+            this.enclaveDataVolMountpoint,
+            containerCreationConfig.getKurtosisVolumeMountpoint());
+        log.trace("New service successfully registered with Kurtosis API");
+
+        log.trace("Loading static files into new service namespace...");
+        const usedStaticFilesMap: Map<string, boolean> = containerCreationConfig.getUsedStaticFiles();
+
+        const usedStaticFiles: Set<string> = new Set();
+        for (let usedStaticFilesId in usedStaticFilesMap) {
+            usedStaticFiles.add(usedStaticFilesId);
+        }
+
+        const resultLoadStaticFiles: Result<Map<string, string>, Error> = await serviceContext.loadStaticFiles(usedStaticFiles); 
+        if (!resultLoadStaticFiles.isOk()) {
+            return err(resultLoadStaticFiles.error);
+        }
+        const staticFileAbsFilepathsOnService: Map<string, string> = resultLoadStaticFiles.value;
+        log.trace("Successfully loaded static files");
+
+        log.trace("Initializing generated files...");
+        const filesToGenerate: Set<string> = new Set();
+        for (let fileId in containerCreationConfig.getFileGeneratingFuncs()) {
+            filesToGenerate.add(fileId);
+        }
+        const resultGenerateFiles: Result<Map<string, GeneratedFileFilepaths>, Error> = await serviceContext.generateFiles(filesToGenerate);
+        if (!resultGenerateFiles.isOk()) {
+            return err(resultGenerateFiles.error);
+        }
+        const generatedFileFilepaths: Map<string, GeneratedFileFilepaths> = resultGenerateFiles.value;
+        const generatedFileAbsFilepathsOnService: Map<string, string> = new Map();
+        for (let [fileId, initializingFunc] of containerCreationConfig.getFileGeneratingFuncs().entries()) {
+
+            if (!generatedFileFilepaths.has(fileId)) {
+                return err(new Error("Needed to initialize file for file ID " + fileId +  ", but no generated file filepaths were " +
+                "found for that file ID; this is a Kurtosis bug"));
+            }
+            const filepaths: GeneratedFileFilepaths = generatedFileFilepaths[fileId];
+
+            const promiseOpenFp: Promise<ResultAsync<number, Error>> = new Promise((resolve, _unusedReject) => {
```

Comment:
Just for the future, I'd prefer if we use more natural language `openFpPromise` and `generateFilesResponse` rather than `promiseOpenFp` and `repsonseGenerateFiles`. No need to change it for this PR, but just for the rest of the changes we made.